src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.ts

```diff
@@ -21,62 +21,92 @@ export const useRespondentAnswers = () => {
   }: FetchAnswers) => {
     if (!appletId || !respondentId) return;
+
     try {
       const {
-        startDate,
+        startDate: formStartDate,
         endDate: formEndDate,
         startTime: formStartTime,
         endTime: formEndTime,
-        identifier,
+        identifier: formIdentifier,
         filterByIdentifier: formFilterByIdentifier,
         versions: formVersions,
         identifiers,
       } = getValues();
-      const endDate = providedEndDate ?? formEndDate;
       const startTime = providedStartTime ?? formStartTime;
       const endTime = providedEndTime ?? formEndTime;
       const filterByIdentifier = providedFilterByIdentifier ?? formFilterByIdentifier;
       const versions = providedVersions ?? formVersions;
+      const identifier = providedIdentifier ?? formIdentifier;
+
+      const { selectedIdentifiers, recentAnswerDateString } =
+        getIdentifiers({
+          filterByIdentifier,
+          filterIdentifiers: identifier,
+          identifiers,
+        }) ?? {};
+
+      const {
+        identifierAnswerStartDate,
+        identifierAnswerEndDate,
+        activityAnswerStartDate,
+        activityAnswerEndDate,
+      } =
+        processIdentifiersChange({
+          setValue,
+          isIdentifiersChange,
+          adjustStartEndDates: !!(filterByIdentifier && identifier?.length),
+          recentIdentifiersAnswerDate: recentAnswerDateString,
+          activityLastAnswerDate: activity.lastAnswerDate,
+        }) ?? {};
 
-      const selectedIdentifiers = getIdentifiers(filterByIdentifier, identifier, identifiers);
+      const startDate = identifierAnswerStartDate ?? activityAnswerStartDate ?? formStartDate;
+      const endDate =
+        identifierAnswerEndDate ??
+        activityAnswerEndDate ??
+        providedEndDate ??
+        formEndDate ??
+        addDays(startDate, 1);
+      const fromDatetime = getDateISO(startDate, startTime);
+      const toDatetime = getDateISO(endDate, endTime);
 
-      const result = await getAnswersApi({
+      const {
+        data: { result: encryptedAnswers },
+      } = await getAnswersApi({
         appletId,
         activityId: activity.id,
         params: {
           respondentId,
-          fromDatetime: getDateISO(startDate, startTime),
-          toDatetime: getDateISO(endDate || addDays(startDate, 1), endTime),
+          fromDatetime,
+          toDatetime,
           emptyIdentifiers: !filterByIdentifier || !selectedIdentifiers?.length,
           identifiers: selectedIdentifiers,
           versions: versions.map(({ id }) => id),
         },
       });
 
-      const encryptedAnswers = result.data.result;
-      const decryptedAnswers = [];
-
-      for await (const encryptedAnswer of encryptedAnswers) {
-        const { userPublicKey, answer, items, itemIds, ...rest } = encryptedAnswer;
-        const decryptedAnswer = (
-          await getDecryptedActivityData({
+      const decryptedAnswers = await Promise.all(
```

Comment:
Since some of encrypted answers can be failed while decryption (transfer ownership reason or whatever reason), then it's better to filter out successfully decrypted answers, otherwise the Promise was failed entirely:
```suggestion
      const decryptedAnswers = await Promise.allSettled(
```

Reply:
I'll return the previous version with for await (const ... of ...).